### PR TITLE
Fix warnings of ping client in vm_virtio_net example

### PR DESCRIPTION
**Pull request by tibersam**
Trivial fixes of type warnings to allow for compilation with Werror.

### CODE REVIEW

File: apps/Arm/vm_virtio_net/components/PingClient/ping_client.c [hunk 1]
```diff
@@ -90,9 +90,11 @@ void print_ip_packet(void *ip_buf, size_t ip_length)
 
 int create_arp_req_reply(char *recv_data, size_t recv_data_size)
 {
-    unsigned char reply_buffer[ETHERMTU];
+    char reply_buffer[ETHERMTU];
 
-    struct ethhdr *rcv_req = (struct ethhdr *) recv_data;
+    //---------------------------------
+    //| ethhdr | ether_arp            |
+    //---------------------------------
```
Comment (axel-h) [inline apps/Arm/vm_virtio_net/components/PingClient/ping_client.c:97]: Documentation improvement are always welcome, but please make this a separate commit as it's not related to fixing the type errors

File: apps/Arm/vm_virtio_net/components/PingClient/ping_client.c [hunk 2]
```diff
@@ -203,12 +205,12 @@ void handle_send_callback(virtqueue_driver_t *vq)
     size_t buf_size = 0, wr_len = 0;
     vq_flags_t flag;
     virtqueue_ring_object_t handle;
-    if (!virtqueue_get_used_buf(vq, &handle, &wr_len)) {
+    if (!virtqueue_get_used_buf(vq, &handle,(uint32_t *) &wr_len)) {
```
Comment (axel-h) [inline apps/Arm/vm_virtio_net/components/PingClient/ping_client.c:208]: We can avoid the cast when changing the base type of the variables. Seems the long-term fix migh be changing the siganture of `virtqueue_get_used_buf()`to use `size_t`

File: apps/Arm/vm_virtio_net/components/PingClient/ping_client.c [hunk 3]
```diff
@@ -203,12 +205,12 @@ void handle_send_callback(virtqueue_driver_t *vq)
...
         ZF_LOGE("Client virtqueue dequeue failed");
         return;
     }
 
-    while (camkes_virtqueue_driver_gather_buffer(vq, &handle, &buf, &buf_size, &flag) >= 0) {
+    while (camkes_virtqueue_driver_gather_buffer(vq, &handle, &buf,(unsigned int *) &buf_size, &flag) >= 0) {
```
Comment (axel-h) [inline apps/Arm/vm_virtio_net/components/PingClient/ping_client.c:213]: Same here, it's better to avoid casting if we can use the proper variable type.

**Comment by tibersam** [comment]
Updated branch, but the system seems to not have picket it up

**Comment by axel-h** [comment]
> Updated branch, but the system seems to not have picked it up

Indeed, that is a bit odd, the commits are there. I've pushed again and now it seems to work. Don't know what that was, seem a github internal problem maybe?

Please address the Gitlint warnings, it's wants shorter linked in the commit title.

File: apps/Arm/vm_virtio_net/components/PingClient/ping_client.c (hunk 1 above)
Comment (axel-h) [reply apps/Arm/vm_virtio_net/components/PingClient/ping_client.c:97]: Let's give it a bit more context:
```suggestion
    // recv_data holds a packet of the form:
    // +--------+-----------+
    // | ethhdr | ether_arp |
    // +--------+-----------+    
```

**Review by kent-mcleod — APPROVED** [review]

**Comment by axel-h** [comment]
There's a type in the commit message "Remove unused variabl warning"  -> "variable"
